ipi/utils/nmtransform.py

```diff
@@ -364,24 +368,63 @@ def __init__(self, nbeads, natoms, open_paths=None):
             import pyfftw
 
             info("Import of PyFFTW successful", verbosity.medium)
-            self.qdummy = pyfftw.n_byte_align_empty((nbeads, 3 * natoms), 16, "float32")
+            self.qdummy = pyfftw.n_byte_align_empty(
+                (nbeads, 3 * natoms),
+                16,
+                "float32" if self.single_precision else "float64",
+            )
             self.qnmdummy = pyfftw.n_byte_align_empty(
-                (nbeads // 2 + 1, 3 * natoms), 16, "complex64"
+                (nbeads // 2 + 1, 3 * natoms),
+                16,
+                "complex64" if self.single_precision else "complex128",
             )
-            self.fft = pyfftw.FFTW(
-                self.qdummy, self.qnmdummy, axes=(0,), direction="FFTW_FORWARD"
+
+            pyfftw.config.NUM_THREADS = self.n_threads
+
+            self.pyfftw_fw = pyfftw.FFTW(
+                self.qdummy,
+                self.qnmdummy,
+                axes=(0,),
+                direction="FFTW_FORWARD",
+                threads=self.n_threads,
             )
-            self.ifft = pyfftw.FFTW(
-                self.qnmdummy, self.qdummy, axes=(0,), direction="FFTW_BACKWARD"
+            self.pyfftw_bw = pyfftw.FFTW(
+                self.qnmdummy,
+                self.qdummy,
+                axes=(0,),
+                direction="FFTW_BACKWARD",
+                threads=self.n_threads,
             )
+
+            # wrapping these calls into functions has negligible overhead and makes profiling easier
+            def call_fft(self):
+                self.pyfftw_fw()
+
+            def call_ifft(self):
+                self.pyfftw_bw()
+
+            self.fft = lambda: call_fft(self)
+            self.ifft = lambda: call_ifft(self)
         except ImportError:  # Uses standard numpy fft library if nothing better
             # is available
             info(
-                "Import of PyFFTW unsuccessful, using NumPy library instead",
-                verbosity.medium,
+                "Import of PyFFTW unsuccessful, using NumPy library instead. ",
+                verbosity.low,
+            )
+            if self.nbeads > 1:
+                warning(
+                    "Install PyFFTW to avoid slow normal-modes integration",
```

Comment:
Do we want to make PyFFTW a dependency of i-PI so that people default to using a faster library?

Reply:
IDK, no strong feelings. I like that i-PI has really minimal dependencies